packages/react-components/src/hooks/useAlarms/utils/updateAlarmValues.ts

```diff
@@ -90,6 +98,18 @@ export const updateAlarmThresholdData = (
   if (!alarm.models) return alarm;
 
   const currentData = alarm.thresholds ?? [];
+
+  /**
+   * If there is a static threshold then that should be used as the alarm threshold
+   **/
+  if (
```

Comment:
I think the root of the problem is that we are filtering out data outside the viewport when we still need to show a datapoint if its the latest before the end of the viewport. So instead of this change let's look into fixing the filtering for alarm state and threshold data.